langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/transport/http/HttpMcpTransport.java

```diff
@@ -116,18 +122,32 @@ private JsonNode executeAndWait(Request request, Long id) {
 
     private EventSource startSseChannel(boolean logResponses) {
         Request request = new Request.Builder().url(sseUrl).build();
-        CompletableFuture<Void> initializationFinished = new CompletableFuture<>();
+        CompletableFuture<String> initializationFinished = new CompletableFuture<>();
         SseEventListener listener = new SseEventListener(pendingOperations, logResponses, initializationFinished);
         EventSource eventSource = EventSources.createFactory(client).newEventSource(request, listener);
-        // wait for the SSE channel to be created, throw an exception if that failed
+        // wait for the SSE channel to be created, receive the POST url from the server, throw an exception if that
+        // failed
         try {
-            initializationFinished.get();
+            int timeout = client.callTimeoutMillis() > 0 ? client.callTimeoutMillis() : Integer.MAX_VALUE;
```

Comment:
Is it a good idea to have `MAX_VALUE` as a timeout?

Reply:
I could instead create an if statement and call the `get()` method without a timeout, the code will look a bit more complicated, but the behavior will basically be the same.

Reply:
I meant that maybe we should specify a bit more meaningful default value for the timeout, perhaps 60-90 seconds? `MAX_VALUE` seems like "wait for it forever"

Reply:
The default is 60 seconds (this is just to handle the case that somebody explicitly specifies the timeout to be 0)

Reply:
Ah, hold on, you may actually be right that we don't have a default. Ok I'll add a default as well, but we have to keep handling the case that somebody specifies it to be zero explicitly (which should mean "no timeout", but calling `completableFuture.get(0)` would throw an exception immediately)

Reply:
I've added a 60 second timeout, right now it's just one value for callTimeout, connectTimeout, readTimeout, writeTimeout. Do you prefer having four separate values?

Reply:
@jmartisk thank you! I think it is ok to have a single value for now